### PR TITLE
Dedupe syn, quote and unicode-xid

### LINKED ISSUE #19786: Update style-derive to syn 0.12
Potentially this could allow builds of servo on nightly to get the improved error messages and other semantics which come from the new span system and `proc-macro` APIs.

`synstructure` has been updated already, although the API has changed since the last version which style-derive used. If there are any APIs or new features which you need in `synstructure` to help port the derives over, let me know. Ideally I'd like to make the `cg` module in style-derive unnecessary, as the features you need are in `synstructure`.

cc @nox

### LINKED ISSUE #20497: Tracking issue for syn 0.13 + quote 0.5 in Servo and dependencies
Do avoid increasing the number versions of syn in Servo’s dependency graph, we should coordinate landing this in various repository at around the same time. So far:

* [x] https://github.com/servo/html5ever/pull/341
* [x] https://github.com/servo/rust-cssparser/pull/217
* [x] https://github.com/servo/string-cache/pull/203
* [x] FIXME: add other affected crates

Ideally, before anything is merged we should have a branch of Servo that uses `[patch.crates-io]` in the root `Cargo.toml` to use all of the relevant PR branches, and check that everything builds with minimal duplication (crates with the same name) in `Cargo.lock`.

### LINKED ISSUE #20609: Build error with custom Webrender
Trying to build Servo with custom Webrender and Webrender_Api results in 2 build errors. 

```
error[E0432]: unresolved import `webrender_api::ScrollEventPhase`
  --> components/compositing/compositor.rs:39:37
   |
39 | use webrender_api::{LayoutVector2D, ScrollEventPhase, ScrollLocation};
   |                                     ^^^^^^^^^^^^^^^^ no `ScrollEventPhase` in the root

error[E0204]: the trait `Copy` may not be implemented for this type
   --> components/compositing/compositor.rs:193:17
    |
193 | #[derive(Clone, Copy)]
    |                 ^^^^
...
202 |     phase: ScrollEventPhase,
    |     ----------------------- this field does not implement `Copy`

error: aborting due to 2 previous errors

```

Looks like `ScrollEventPhase` was removed in https://github.com/servo/webrender/pull/2629.

### PR DESCRIPTION
**Pull request by Eijebong**
Dupe num-traits instead...

Fixes #20609 
Fixes #20497
Fixes #19786
Closes #20564

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20611)

### CODE REVIEW

**Comment by Eijebong** [comment]
@glennw Could you have a look at the webrender changes ?
This is still missing some releases to be complete but I wanted to be sure that all the webrender related changes were correct

File: components/layout/display_list/webrender_helpers.rs [hunk 1]
```diff
@@ -225,6 +225,7 @@ impl WebRenderDisplayItemConverter for DisplayItem {
 
                 builder.push_stacking_context(
                     &webrender_api::LayoutPrimitiveInfo::new(stacking_context.bounds),
+                    Some(scrolling_id),
```
Comment (Eijebong) [inline components/layout/display_list/webrender_helpers.rs]: It's the only `ClipId` I found in that function. Maybe it should be None ?

Comment (glennw) [reply components/layout/display_list/webrender_helpers.rs]: Yup, should be None for now, to retain existing behavior.

**Comment by Eijebong** [comment]
@bors-servo try

Let's see how much stuff I broke

**Comment by Eijebong** [comment]
@bors-servo try

Mac build should be fixed

**Comment by Eijebong** [comment]
@bors-servo try

This time it should be ok >_<

**Comment by glennw** [comment]
```
   Compiling style_tests v0.0.1 (file:///home/servo/buildbot/slave/android/build/tests/unit/style)
error[E0432]: unresolved import `libc`
  --> components/gfx/platform/mod.rs:19:9
   |
19 |     use libc::c_char;
   |         ^^^^ Maybe a missing `extern crate libc;`?
```

**Pull request closed without merging**